scispacy/base_project_code.py

```diff
@@ -0,0 +1,109 @@
+from typing import Optional, Callable
+from pathlib import Path

```

Comment:
This file is missing type hints in some places, but otherwise I like how straightforward adding in our custom readers etc is. This was my main concern about how we would use the projects stuff. Nice one for figuring it out!